tests/lib/c_lib/common/src/main.c

```diff
@@ -1166,6 +1166,9 @@ ZTEST(libc_common, test_time_ctime)
 	char buf[26] = {0};
 	time_t test1 = 1718260000;
 
+#ifdef CONFIG_NATIVE_LIBC
+	setenv("TZ", "UTC", 1);
```

Comment:
LGTM - it might be helpful to have a comment here as to why this is needed (which IIRC, is because the Zephyr common libc `ctime_r()` implementation assumes UTC timezone, since timezone support is usually beyond the scope of most devices).

Reply:
That's not completely accurate; newlib and picolibc both support timezones, but those depend on something actually setting the TZ environment variable within the runtime. CONFIG_NATIVE_LIBC is the right test because that's the only library which can inherit external system settings and find a non-UTC timezone.

Reply:
Oh wow - I guess that's something I haven't considered in embedded for a while 😅